brushfire-core/src/main/scala/com/stripe/brushfire/AnnotatedTree.scala

```diff
@@ -27,6 +27,22 @@ object LeafNode {
 }
 
 case class AnnotatedTree[K, V, T, A: Semigroup](root: Node[K, V, T, A]) {
+
+  private var sumTargetsCache: Option[T] = None
```

Comment:
Not loving this but it can't just be lazy because it needs the monoid. Maybe we should lift the `Monoid[T]` to be something the tree carries around?